x-pack/solutions/observability/plugins/observability/public/pages/alert_details/hooks/use_discover_url.ts

```diff
@@ -35,10 +49,50 @@ const isViewInDiscoverSupportedRuleType = (
   );
 };
 
+const SYNTHETICS_TEMP_DATA_VIEW: DataViewSpec = {
+  title: 'synthetics-*',
+  timeFieldName: '@timestamp',
+};
+/**
+ * For certain rule types, we create a temporary data view.
+ * Otherwise, returns undefined, and an existing saved data view must be specified.
+ * @param rule an Observability alerting rule
+ * @returns A temporary data view spec, or undefined
+ */
+const getCustomDataViewParams = (rule?: Rule): DataViewSpec | undefined => {
+  switch (rule?.ruleTypeId) {
+    case SYNTHETICS_TLS_RULE:
+    case SYNTHETICS_STATUS_RULE:
+      return SYNTHETICS_TEMP_DATA_VIEW;
+    default:
+      return undefined;
+  }
+};
+
 const getLocatorParamsMap: Record<
   (typeof viewInDiscoverSupportedRuleTypes)[number],
   (rule: Rule) => DiscoverAppLocatorParams
 > = {
+  [SYNTHETICS_STATUS_RULE]: (rule): DiscoverAppLocatorParams => {
+    const params = rule.params as SyntheticsMonitorStatusRuleParams;
+    const query = syntheticsMonitorStatusAlertParamsToKqlQuery(params);
+    return {
+      query: {
+        language: 'kuery',
+        query,
+      },
+    };
+  },
+  [SYNTHETICS_TLS_RULE]: (rule) => {
+    const params = rule.params as TLSRuleParams;
+    const query = syntheticsTlsAlertParamsToKqlQuery(params);
+    return {
+      query: {
+        language: 'kuery',
+        query,
+      },
+    };
+  },
```

Comment:
I'd like this method to return all the params needed to get the redirect url, how do you feel about returning the `dataViewSpec` here?

Reply:
That is fine. I had to solve the data spec as part of this implementation, and as I handled that piece separately from building the filters, it did not cross my mind to combine them.